votrfront/js/sorting.js

```diff
@@ -73,3 +74,184 @@ export function sortTable(items, columns, query, queryKey) {
+
+  function handleClick(event) {
+    var index = event.currentTarget.getAttribute('data-index');
+    var {label, prop, process, preferDesc} = columns[index];
+
+    var newOrder = _.without(order, 'a' + index, 'd' + index);
+    newOrder.unshift((
+      order[0] == 'a' + index ? 'd' :
+      order[0] == 'd' + index ? 'a' :
+      preferDesc ? 'd' : 'a') + index);
+
+    navigate({ ...query, [queryKey]: newOrder.join('') });
+  }
+
+  var header = <tr>
+    {columns.map(({label, prop, process, preferDesc, hiddenClass = []}, index) =>
+      <th key={index} data-index={index} onClick={handleClick}
+          className={`${hiddenClass.join(" ")} ` + 'sort ' + (order[0] == 'a' + index ? 'asc' :
+                                order[0] == 'd' + index ? 'desc' : '')}>
+        {label}
+      </th>
+    )}
+  </tr>;
+
+  return [items, header];
+};
+
+function toggleInfo(index) {
+  const opened = JSON.parse(LocalSettings.get("openedRows")) || {};
+  opened[index] = !opened[index];
+  LocalSettings.set("openedRows", JSON.stringify(opened));
+}
+
+function expandAll(items) {
+  const opened = items.reduce((acc, curr, idx) => ({...acc, [idx]: true}), {});
+  LocalSettings.set("openedRows", JSON.stringify(opened));
+}
+
+// TODO: better names
+function collapseAll() {
+  LocalSettings.set("openedRows", JSON.stringify({}));
+}
+
+function isOpened(index) {
+  return (JSON.parse(LocalSettings.get("openedRows")) || {})[index];
+}
+
+export const SortableTable = ({
+  items,
+  columns,
+  queryKey,
+  withButtons,
+  footer,
+  message
+}) =>
+  queryConsumer(query => {
+    // TODO change to sortTable
+    const [sortedItems, header] = sortTableCopyForObjectCol(
+      items,
+      columns,
+      query,
+      queryKey
+    );
+
+    const className =
+      "table table-condensed table-bordered table-striped table-hover" +
+      (withButtons ? " with-buttons-table" : "");
+
+    const notExpandable = columns.reduce(
+      (acc, col) =>
+        acc.filter(item => !(col.hiddenClass && col.hiddenClass.includes(item))),
+      ["hidden-xs", "hidden-sm", "hidden-md", "hidden-lg"]
+    );
+
+    const rows = [];
+
+    sortedItems.forEach((item, index) => {
+      rows.push(
+        <tr key={index} onClick={() => toggleInfo(index)}>
+          {columns.map(
+            ({
+              label,
+              prop,
+              process,
+              hiddenClass = [],
+              cell,
+              colProps,
+              expansionMark
+            }) => (
+              <td
+                key={label}
+                className={hiddenClass.join(" ")}
+                {...(colProps instanceof Function ? colProps(item) : {})}
+              >
+                {expansionMark && (
+                  <span
+                    className={notExpandable.join(" ")}
+                    style={{ fontWeight: "bold", float: "right" }}
+                  >
+                    {isOpened(index) ? "-" : "+"}
```

Comment:
Tu by som buď dal "[-]" a "[+]", ktoré sú snáď kúsok jasnejšie, alebo ak sa nudíš spravil nejakú CSS mágiu aby boli vyložene v štvorčeku. (Bonus: nastaviť fixnú šírku, aby sa neposúval textový obsah bunky ak použiješ môj návrh dať to doľava.)